go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/hashicorp/terraform-plugin-framework-timeouts
 
-go 1.23.0
-
-toolchain go1.23.7
+go 1.24.0
```

Comment:
TIL, the toolchain will just use the latest Go patch version, perhaps we should adopt that everywhere 🤔